report/renderer/api.js

```diff
@@ -12,6 +12,9 @@ import {ReportUIFeatures} from '../renderer/report-ui-features.js';
 import {CategoryRenderer} from './category-renderer.js';
 import {DetailsRenderer} from './details-renderer.js';
 
+/** @type {WeakMap<HTMLElement, ReportUIFeatures>} */
+const rootElToReportUIFeatures = new WeakMap();
```

Comment:
Could you add some tests for this?

Storing the renderer state in globals can lead to lingering instances between test cases. Probably won't be a problem here since you use a map, but should still keep that in mind.

Reply:
What sort of test? Every call to render will make a new element, and so a new key->value pair in this map.

ahhh this tile has _zero_ tests eh? kinda makes since, they api surface and methods are pretty minimal, and each individual component is pretty well tested. Not quite certain what tests should be added that wouldn't just be duplicates of existing tests.

one test I thought of is just doing `renderReport` and then `addButton` then confirming the button is in the `el`. And doing a second report render and confirming the same.

Reply:
I was thinking of tests that verify the association between a `rootEl` and an instance of `ReportUIFeatures`. The one you mentioned would work.